src/ring/rack.clj

```diff
@@ -25,8 +25,9 @@
 
       def responsify(output)
         begin
-          return output[0].to_java if output.respond_to?(:size) && output.size > 1
-          output.each{|s| s.to_java}
+          retval = []
+          output.each {|s| retval.push(s)}
+          retval
```

Comment:
Ok, but you removed my optimization (no array allocation needed if result size is 1) :(